SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestEntityInteration/EntityInteractionTest.cpp

```diff
@@ -0,0 +1,144 @@
+			for (AActor* Actor : TestActors)
+			{
+				if (AEntityInteractionTestActor* TestActor = Cast<AEntityInteractionTestActor>(Actor))
+				{
+					if (ensure(TestActor->Index < 2))
+					{
+						if (TestActor->HasAuthority())
+						{
+							LocalActors[TestActor->Index] = TestActor;
+						}
+						else
+						{
+							RemoteActors[TestActor->Index] = TestActor;
+						}
+					}
+				}
+			}
+			for (auto Actor : LocalActors)
+			{
+				AssertTrue(Actor != nullptr, TEXT("Missing test actors"));
+			}
+			for (auto Actor : RemoteActors)
+			{
+				AssertTrue(Actor != nullptr, TEXT("Missing test actors"));
+			}
+			FinishStep();
+		},
+		nullptr);
+
+	AddStep(
+		"Send RPCs", FWorkerDefinition::AllServers, nullptr,
+		[this]() {
+			AActor::ExecuteWithNetWriteFence(RemoteActors[0], LocalActors[0], &AEntityInteractionTestActor::TestNetWriteFence, NumSteps);
+			EXPECT_IMMEDIATE(LocalActors[0], AEntityInteractionTestActor::s_NetWriteFenceName, TEXT("Remote dependent"));
+
+			AActor::ExecuteWithNetWriteFence(LocalActors[1], LocalActors[0], &AEntityInteractionTestActor::TestNetWriteFence, NumSteps);
+			EXPECT_DELAYED(LocalActors[0], AEntityInteractionTestActor::s_NetWriteFenceName, TEXT("Local dependent"));
+
+			LocalActors[1]->SendCrossServerRPC(LocalActors[0], &AEntityInteractionTestActor::TestReliable, NumSteps);
+			EXPECT_IMMEDIATE(LocalActors[0], AEntityInteractionTestActor::s_ReliableName, TEXT("Local Receiver"));
+
+			LocalActors[1]->SendCrossServerRPC(RemoteActors[0], &AEntityInteractionTestActor::TestReliable, NumSteps);
+			EXPECT_DELAYED(RemoteActors[0], AEntityInteractionTestActor::s_ReliableName, TEXT("Remote Receiver"));
+
+			LocalActors[0]->TestUnreliable(NumSteps);
+			EXPECT_IMMEDIATE(LocalActors[0], AEntityInteractionTestActor::s_UnreliableName, TEXT("Local Receiver"));
+			RemoteActors[0]->TestUnreliable(NumSteps);
+			EXPECT_DELAYED(RemoteActors[0], AEntityInteractionTestActor::s_UnreliableName, TEXT("Remote Receiver"));
+
+			LocalActors[0]->TestUnordered(NumSteps);
+			EXPECT_IMMEDIATE(LocalActors[0], AEntityInteractionTestActor::s_UnorderedName, TEXT("Local Receiver"));
+			RemoteActors[0]->TestUnordered(NumSteps);
+			EXPECT_DELAYED(RemoteActors[0], AEntityInteractionTestActor::s_UnorderedName, TEXT("Remote Receiver"));
+
+			LocalActors[1]->SendCrossServerRPC(LocalActors[0], &AEntityInteractionTestActor::TestNoLoopback, NumSteps);
+			EXPECT_DELAYED(LocalActors[0], AEntityInteractionTestActor::s_NoLoopbackName, TEXT("Local Receiver"));
+
+			LocalActors[1]->SendCrossServerRPC(RemoteActors[0], &AEntityInteractionTestActor::TestNoLoopback, NumSteps);
+			EXPECT_DELAYED(RemoteActors[0], AEntityInteractionTestActor::s_NoLoopbackName, TEXT("Remote Receiver"));
+
+			FinishStep();
+		},
+		nullptr, 5.0);
+
+	AddStep(
+		"Check delayed RPCs", FWorkerDefinition::AllServers,
+		[this]() {
+			if (LocalActors[0]->Steps.Num() == 10)
```

Comment:
Why 10?

Reply:
That's the number of sent RPCs. Could make it something other than a magic number.

Reply:
Moving it into a named variable would be good.

Reply:
Using NumSteps instead